src/game/process/cmd/teams.js

```diff
@@ -0,0 +1,63 @@
+import { TEAMS } from '../../common/const.mjs';
+
+export function assignToTeam(state, walletAddress) {
+  const teamsConfig = state.teamsConfig;
+  const teams = teamsConfig.teams;
+  const amount = teamsConfig.amount;
+  if (teams.length < amount) {
+    const id = teams.length;
+    const team = Object.values(TEAMS).find((c) => c.id == id);
+    teams[id] = {
+      id,
+      color: team.color,
+      name: team.name,
+      players: [walletAddress],
+      points: 0,
+    };
+    return teams[id];
+  } else {
+    for (let i = 1; i < teams.length; i++) {
+      if (teams[i].players.length < teams[i - 1].players.length) {
```

Comment:
also, instead of this logic I would simply remember in the state what was the idx of the last team that the player was added to (initialize it in the initState to sth like `state.lastTeamIdx = -1`). 
The code could be then simplified to sth like
```
state.lastTeamIdx = state.lastTeamIdx + 1 > teams.length - 1 ? 0 : ++state.lastTeamIdx;
teams[state.lastTeamIdx].players.push(walletAddress);
```